src/pages/Lend/Filter/FilterSections/LocationMulti/LocationMultiRefinements.vue

```diff
@@ -0,0 +1,224 @@
+<template>
+	<div class="location-refinements-controller">
+		<div
+			v-for="item in items"
+			:key="item.label"
+		>
+			<div
+				v-if="item.isRegion"
+				class="region"
+				:class="{open: isRegionOpen(item.region)}"
+			>
+				<div
+					class="region-title-row"
+					@click="toggleRegionOpenState(item.region)"
+				>
+					<kv-icon
+						name="large-chevron"
+						class="hierarchical-menu-item-chevron"
+					/>
+					<div
+						class="region-label"
+					>
+						{{ item.label }} ({{ item.count }})
+					</div>
+				</div>
+				<div
+					v-if="isRegionOpen(item.region)"
+					class="select-all-toggle-container"
+				>
+					<div
+						v-if="!item.allCountriesSelected"
+						@click="selectAllCountries(item.countries, item.region)"
+					>
+						Select all
+					</div>
+					<div
+						v-else
+						@click="deselectAllCountries(item.countries, item.region)"
+					>
+						Deselect all
+					</div>
+				</div>
+			</div>
+			<kv-controlled-checkbox
+				v-else-if="isRegionOpen(item.region)"
+				class="country"
+				:checked="isCountryChecked(item.isRefined, item.region)"
+				:label="`${item.label} (${item.count})`"
+				@checkbox-input="handleCountryCheckboxInput(item)"
+			/>
+		</div>
+	</div>
+</template>
+
+<script>
+import KvIcon from '@/components/Kv/KvIcon';
+import KvControlledCheckbox from '@/components/Kv/KvControlledCheckbox';
+
+export default {
+	components: {
+		KvControlledCheckbox,
+		KvIcon,
+	},
+	data() {
+		return {
+			regionMap: {
+				Africa: {
+					open: false,
+					temporaryRefinement: null,
+				},
+				Asia: {
+					open: false,
+					temporaryRefinement: null,
+				},
+				'Central America': {
+					open: false,
+					temporaryRefinement: null,
+				},
+				'Eastern Europe': {
+					open: false,
+					temporaryRefinement: null,
+				},
+				'Middle East': {
+					open: false,
+					temporaryRefinement: null,
+				},
+				'North America': {
+					open: false,
+					temporaryRefinement: null,
+				},
+				Oceania: {
+					open: false,
+					temporaryRefinement: null,
+				},
+				'South America': {
+					open: false,
+					temporaryRefinement: null,
+				},
+			},
+		};
+	},
+	props: {
+		items: {
+			type: Array,
+			default: () => [],
+		},
+		refine: {
+			type: Function,
+			default: () => {},
+		},
+	},
+	methods: {
+		handleCountryCheckboxInput(item) {
+			this.refine(item.value);
+		},
+		refineAllCountries(countries, refineTo) {
+			countries.forEach(({ isRefined, value }) => {
+				if (isRefined !== refineTo) {
+					this.refine(value);
+				}
+			});
+		},
+		selectAllCountries(countries, region) {
+			if (this.regionMap[region].temporaryRefinement !== null) {
+				return;
```

Comment:
Added a debounce of 500ms to ignore double clicks and allow Algolia time to process